### PR TITLE
Add support for retrieving the JSON version of the plan

### PR DESCRIPTION
**Pull request by sfc-gh-ahess**
Implements the API function "JSON Output" for the plan and returns it as a byte array.

### CODE REVIEW

File: plan.go [hunk 1]
```diff
@@ -134,3 +138,27 @@ func (s *plans) Logs(ctx context.Context, planID string) (io.Reader, error) {
...
+	if !validStringID(&planID) {
+		return nil, errors.New("invalid value for plan ID")
+	}
+
+	u := fmt.Sprintf("plans/%s/json-output", url.QueryEscape(planID))
+	// TODO: FInd a way to not have to use our own code to do this - maybe extend the API?
+	req, err := s.client.newRequest("GET", u, nil)
+	if err != nil {
+		return nil, fmt.Errorf("tfe: unable to get json plan %v", err)
+	}
```
Comment (radditude) [inline plan.go:154]: When handling request errors elsewhere, we generally return the error from the API directly:

```
	if err != nil {
		return nil, err
	}
```

Is there a special reason we need extra error text here?

File: plan.go [hunk 2]
```diff
@@ -134,3 +138,27 @@ func (s *plans) Logs(ctx context.Context, planID string) (io.Reader, error) {
...
+
+
+// Provide the JSON version of the plan from the TFE API.
+func (s *plans) JsonOutput(ctx context.Context, planID string) ([]byte, error) {
+	if !validStringID(&planID) {
+		return nil, errors.New("invalid value for plan ID")
+	}
+
+	u := fmt.Sprintf("plans/%s/json-output", url.QueryEscape(planID))
+	// TODO: FInd a way to not have to use our own code to do this - maybe extend the API?
```
Comment (radditude) [inline plan.go:150]: I'm not quite sure I understand what this TODO comment is suggesting - could you clarify?

**Review by radditude — CHANGES_REQUESTED** [review]
Thank you for splitting these out!

I'd love to add some tests for `Plans.JsonOutput`. You can see an example of tests for a similar method in [`TestStateVersionDownload`](https://github.com/hashicorp/go-tfe/blob/master/state_version_test.go#L386), though I wouldn't recommend comparing the full output against a static fixture file like we do there, since plan output will vary depending on the Terraform version used in the workspace.

Instead, I'd just check that the output is not empty and contains some string representative of JSON plan output, like `"address": "null_resource.foo"`, to confirm that retrieving the plan output was successful.

File: plan.go (hunk 1 above)
Comment (sfc-gh-ahess) [reply plan.go]: I've updated this to conform to your design preferences.

File: plan.go (hunk 2 above)
Comment (sfc-gh-ahess) [reply plan.go]: Sorry, it was left over from before, when I was doing it a different way.  I've removed it now.

**Comment by sfc-gh-ahess** [comment]
I'm just implementing a missing part of your API here, unfortunately it is beyond my scope of work to do any additional development on this project.  Please feel free to take my code and add any additional changes that you require.

**Comment by radditude** [comment]
Thanks for contributing! I'm going to close this out in favor of #187.

**Pull request closed without merging**